manifests/init.pp

```diff
@@ -39,6 +39,7 @@
 
   Optional[Enum['single', 'peruser']] $repomodel              = undef,
   Optional[Enum['readwrite', 'readonly', 'none']] $modulepath = undef,
+  Optional[Enum['yes','no']] $allow_root                       = undef,
```

Comment:
@ipcrm This is essentially the equivalent of the `Boolean` type if you set the default to `false` instead of allowing `undef` as a value. Can you update the code with that type instead and elsewhere as needed? We'll go ahead and merge after that.